VTK keywords are case insensitive

According to [_File Formats for VTK Version 4.2_](https://www.vtk.org/wp-content/uploads/2015/04/file-formats.pdf) (linked to from meshio/README.md) (and to [_The VTK User's Guide_, 11th edition](https://www.kitware.com/products/books/VTKUsersGuide.pdf), §19.3), for ‘Simple Legacy Formats’,
> Keywords are case insensitive

…but `meshio.read` baulks on [membrane.vtk](https://github.com/nschloe/meshio/files/2413803/membrane.tar.gz), generated from by _FreeFem++_ from its `examples++-chapt3/membrane.edp` augmented by
```c++
load "iovtk"
savevtk("membrane.vtk", Th, phi);
```
since that contains at line 4089
```
CELL_DATA 2718
Scalars  Label int 1
```
and so `meshio.read('membrane.vtk')` raises
> AssertionError: Unknown section 'Scalars'.

Paraview reads the file happily.

I think that this should be easy to fix and will attempt to do so presently.

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -125,7 +125,7 @@ def read_buffer(f):
     f.readline()
     f.readline()
 
-    data_type = f.readline().decode("utf-8").strip()
+    data_type = f.readline().decode("utf-8").strip().upper()
     assert data_type in ["ASCII", "BINARY"], "Unknown VTK data type '{}'.".format(
         data_type
     )
@@ -152,10 +152,10 @@ def read_buffer(f):
             continue
 
         split = line.split()
-        section = split[0]
+        section = split[0].upper()
 
         if section == "DATASET":
-            dataset_type = split[1]
+            dataset_type = split[1].upper()
             assert (
                 dataset_type == "UNSTRUCTURED_GRID"
             ), "Only VTK UNSTRUCTURED_GRID supported (not {}).".format(
@@ -165,7 +165,7 @@ def read_buffer(f):
         elif section == "POINTS":
             active = "POINTS"
             num_points = int(split[1])
-            data_type = split[2]
+            data_type = split[2].upper()
             points = _read_points(f, data_type, is_ascii, num_points)
 
         elif section == "CELLS":

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -186,6 +186,10 @@ def read_buffer(f):
             active = "CELL_DATA"
             num_items = int(split[1])
 
+        elif section == "LOOKUP_TABLE":
+            num_items = int(split[2])
+            rgba = numpy.fromfile(f, count=num_items * 4, sep=" ", dtype=float)
+
         elif section == "SCALARS":
             if active == "POINT_DATA":
                 d = point_data

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -188,7 +188,8 @@ def read_buffer(f):
 
         elif section == "LOOKUP_TABLE":
             num_items = int(split[2])
-            rgba = numpy.fromfile(f, count=num_items * 4, sep=" ", dtype=float)
+            data = numpy.fromfile(f, count=num_items * 4, sep=" ", dtype=float)
+            rgba = data.reshape((num_items, 4))  # noqa F841
 
         elif section == "SCALARS":
             if active == "POINT_DATA":

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -165,7 +165,7 @@ def read_buffer(f):
         elif section == "POINTS":
             active = "POINTS"
             num_points = int(split[1])
-            data_type = split[2].upper()
+            data_type = split[2].lower()
             points = _read_points(f, data_type, is_ascii, num_points)
 
         elif section == "CELLS":
@@ -282,7 +282,7 @@ def _read_cell_types(f, is_ascii, num_items):
 
 def _read_scalar_field(f, num_data, split):
     data_name = split[1]
-    data_type = split[2]
+    data_type = split[2].lower()
     try:
         num_comp = int(split[3])
     except IndexError:
@@ -293,7 +293,7 @@ def _read_scalar_field(f, num_data, split):
     assert 0 < num_comp < 5
 
     dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type])
-    lt, _ = f.readline().decode("utf-8").split()
+    lt, _ = f.readline().decode("utf-8").split().upper()
     assert lt == "LOOKUP_TABLE"
     data = numpy.fromfile(f, count=num_data, sep=" ", dtype=dtype)
 
@@ -312,7 +312,7 @@ def _read_vector_field(f, num_data, split):
 
 def _read_tensor_field(f, num_data, split):
     data_name = split[1]
-    data_type = split[2]
+    data_type = split[2].lower()
 
     dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type])
     data = numpy.fromfile(f, count=9 * num_data, sep=" ", dtype=dtype).reshape(-1, 3, 3)
@@ -326,7 +326,7 @@ def _read_fields(f, num_fields, is_ascii):
         name, shape0, shape1, data_type = f.readline().decode("utf-8").split()
         shape0 = int(shape0)
         shape1 = int(shape1)
-        dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type])
+        dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type.lower()])
 
         if is_ascii:
             dat = numpy.fromfile(f, count=shape0 * shape1, sep=" ", dtype=dtype)

--- a/meshio/vtk_io.py
+++ b/meshio/vtk_io.py
@@ -293,8 +293,8 @@ def _read_scalar_field(f, num_data, split):
     assert 0 < num_comp < 5
 
     dtype = numpy.dtype(vtk_to_numpy_dtype_name[data_type])
-    lt, _ = f.readline().decode("utf-8").split().upper()
-    assert lt == "LOOKUP_TABLE"
+    lt, _ = f.readline().decode("utf-8").split()
+    assert lt.upper() == "LOOKUP_TABLE"
     data = numpy.fromfile(f, count=num_data, sep=" ", dtype=dtype)
 
     return {data_name: data}

